jedi/parser/tree.py

```diff
@@ -865,8 +879,13 @@ def is_generator(self):
         return bool(self.yields)
 
     def annotation(self):
+        if self.children[0] == "lambda":
```

Comment:
Instead of checking this here, I think you should overwrite it in `Lambda.annotation`. This is cleaner IMO.